PyInstaller/compat.py

```diff
@@ -889,3 +891,30 @@ def check_requirements():
             raise SystemExit('PyInstaller cannot check for assembly dependencies.\n'
                              'Please install PyWin32.\n\n'
                              'pip install pypiwin32\n')
+
+
+@contextmanager
+def open_pysrc(filename):
+    if getattr(tokenize, 'open', None) is not None:
+        print('USING tokenize')
+        pysrc_f = tokenize.open(filename)
+    else:
+        print('Guessing encoding')
+        from PyInstaller.lib.modulegraph.util import guess_encoding
```

Comment:
ok, I just saw imports occassionally interleaved throughout the source, so I was thinking it's not a problem to use it this way. I will move it to the top.